packages/gatsby/cache-dir/slice.js

```diff
@@ -0,0 +1,112 @@
+import React, { useContext } from "react"
+import { ServerSlice } from "./slice/server-slice"
+import { InlineSlice } from "./slice/inline-slice"
+import { SlicesContext } from "./slice/context"
+
+export function Slice(props) {
+  if (_CFLAGS_.GATSBY_MAJOR === `5` && process.env.GATSBY_SLICES) {
+    // we use sliceName internally, so remap alias to sliceName
+    const internalProps = {
+      ...props,
+      sliceName: props.alias,
+    }
+    delete internalProps.alias
+
+    const slicesContext = useContext(SlicesContext)
+
+    // validate props
+    const propErrors = validateSliceProps(props)
+    if (Object.keys(propErrors).length) {
+      throw new SlicePropsError(
+        slicesContext.renderEnvironment === `browser`,
+        internalProps.sliceName,
+        propErrors
+      )
+    }
+
+    if (slicesContext.renderEnvironment === `server`) {
+      return <ServerSlice {...internalProps} />
+    } else if (slicesContext.renderEnvironment === `browser`) {
+      // in the browser, we'll just render the component as is
+      return <InlineSlice {...internalProps} />
+    } else if (slicesContext.renderEnvironment === `engines`) {
+      // if we're in SSR, we'll just render the component as is
+      return <InlineSlice {...internalProps} />
+    } else {
+      throw new Error(
+        `Slice context "${slicesContext.renderEnvironment}" is not supported.`
+      )
+    }
+  } else {
+    throw new Error(`Slices are only available in Gatsby v5`)
+  }
+}
+
+class SlicePropsError extends Error {
+  constructor(inBrowser, sliceName, propErrors) {
+    const errors = Object.entries(propErrors)
+      .map(([key, value]) => `${key}: "${value}"`)
+      .join(`, `)
+
+    const name = `SlicePropsError`
+    let stack = ``
+    let message = ``
+
+    if (inBrowser) {
+      // They're just (kinda) kidding, I promise... You can still work here <3
+      //   https://www.gatsbyjs.com/careers/
+      const fullStack =
+        React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactDebugCurrentFrame.getCurrentStack()
+
+      // remove the first line of the stack trace
+      const stackLines = fullStack.trim().split(`\n`).slice(1)
+      stackLines[0] = stackLines[0].trim()
+      stack = `\n` + stackLines.join(`\n`)
+
+      // look for any hints for the component name in the stack trace
+      const componentRe = /^at\s+([a-zA-Z0-9]+)/
+      const componentMatch = stackLines[0].match(componentRe)
+      const componentHint = componentMatch ? `in ${componentMatch[1]} ` : ``
+
+      message = `Slice "${sliceName}" was passed props ${componentHint}that are not serializable (${errors}).`
+    } else {
+      // we can't really grab any extra info outside of the browser, so just print what we can
+      message = `${name}: Slice "${sliceName}" was passed props that are not serializable (${errors}). Use \`gatsby develop\` to see more information.`
```

Comment:
causing error:
![image](https://user-images.githubusercontent.com/419821/194328430-d1a9e498-6dec-4662-82cc-f4ffdcbf9bb9.png)

it shows in overlay, however code frame, code location is not right currently
![image](https://user-images.githubusercontent.com/419821/194328353-f82d5616-aa65-4422-bbcd-0cf7aac1b085.png)

Reply:
Something must have changed recently or we added a level of depth in our tree. 

It used to render to the first line of the file containing the error-ing Slice. Should be somewhere in here: https://github.com/gatsbyjs/gatsby/blob/1b94c72b141a6bca1cdcf486d4b67925b76efc78/packages/gatsby/cache-dir/slice.js#L61

Reply:
![image](https://user-images.githubusercontent.com/5430474/194360232-fb7a980b-7887-4c60-b930-59f499e46f94.png)

Reply:
This proves to be difficult to resolve in confident way. The component stack I'm getting is missing file/line number location for a lot of frames (including the component that uses the slice):

```
    at Slice (webpack-internal:///./.cache/slice.js:32:24)
    at main
    at IndexPage
    at PageRenderer (webpack-internal:///./.cache/page-renderer.js:21:84)
    at PageQueryStore (webpack-internal:///./.cache/query-result-store.js:40:30)
    at RouteHandler
    at div
    at oe (webpack-internal:///./node_modules/@gatsbyjs/reach-router/dist/index.modern.mjs:38:9978)
    at re (webpack-internal:///./node_modules/@gatsbyjs/reach-router/dist/index.modern.mjs:38:9765)
    at ie (webpack-internal:///./node_modules/@gatsbyjs/reach-router/dist/index.modern.mjs:38:11249)
    at se (webpack-internal:///./node_modules/@gatsbyjs/reach-router/dist/index.modern.mjs:38:11123)
    at ScrollHandler (webpack-internal:///./node_modules/gatsby-react-router-scroll/scroll-handler.js:36:35)
    at RouteUpdates (webpack-internal:///./.cache/navigation.js:289:32)
    at EnsureResources (webpack-internal:///./.cache/ensure-resources.js:22:30)
    at LocationHandler (webpack-internal:///./.cache/root.js:64:29)
    at eval (webpack-internal:///./node_modules/@gatsbyjs/reach-router/dist/index.modern.mjs:38:8351)
    at H (webpack-internal:///./node_modules/@gatsbyjs/reach-router/dist/index.modern.mjs:38:7249)
    at G (webpack-internal:///./node_modules/@gatsbyjs/reach-router/dist/index.modern.mjs:38:7551)
    at WithErrorBoundary()
    at Q (webpack-internal:///./node_modules/@gatsbyjs/reach-router/dist/index.modern.mjs:38:9142)
    at Root
    at StaticQueryStore (webpack-internal:///./.cache/query-result-store.js:159:32)
    at SliceDataStore (webpack-internal:///./.cache/query-result-store.js:215:32)
    at ErrorBoundary (webpack-internal:///./.cache/fast-refresh-overlay/components/error-boundary.js:24:35)
    at DevOverlay (webpack-internal:///./.cache/fast-refresh-overlay/index.js:124:5)
    at RootWrappedWithOverlayAndProvider
    at App (webpack-internal:///./.cache/app.js:184:80)
```

During debugging I found information about code location in fibers data, but that data isn't actually used for produced component stack:

![image](https://user-images.githubusercontent.com/419821/194502298-83d3cf22-1ab8-4ad5-b70a-a5a4dbfcad72.png)

this might be avenue to get correct location in overlay, but I don't want to keep this PR hostage for this